tstrait/genetic.py

```diff
@@ -0,0 +1,207 @@
+import numba
+import numpy as np
+import pandas as pd
+import tskit
+
+from .base import _check_instance, _check_dataframe, _check_non_decreasing  # noreorder
+
+
+@numba.njit
+def _traversal_genotype(
```

Comment:
As below, ``genotype`` isn't really the right word and we may as well sum the effect sizes directly.